tools/nnyeah/nnyeah/AssemblyComparator/TypeAndMemberMap.cs

```diff
@@ -23,8 +24,48 @@ public class TypeAndMemberMap {
 		public List<string> PropertiesNotPresent { get; init; } = new List<string> ();
 		public Dictionary<string, PropertyDefinition> PropertyMap { get; init; } = new Dictionary<string, PropertyDefinition> ();
 
-		public TypeAndMemberMap ()
+		public ModuleDefinition MicrosoftModule { get; init; }
+
+		public TypeAndMemberMap (ModuleDefinition module)
+		{
+			MicrosoftModule = module;
+		}
+
+		public bool TypeIsNotPresent (string typeName)
+		{
+			return TypesNotPresent.Contains (typeName);
+		}
+
+		public bool TryGetMappedType (string typeName, [NotNullWhen (returnValue: true)] out TypeDefinition? result)
+		{
+			return TypeMap.TryGetValue (typeName, out result);
+		}
+
+		public bool MemberIsNotPresent (string member)
+		{
+			return MethodsNotPresent.Contains (member) ||
+				FieldsNotPresent.Contains (member) ||
+				EventsNotPresent.Contains (member) ||
+				PropertiesNotPresent.Contains (member);
+		}
+
+		public bool TryGetMappedMember (string memberName, [NotNullWhen (returnValue: true)] out IMemberDefinition? member)
 		{
+			if (MethodMap.TryGetValue (memberName, out var method)) {
+				member = method;
+				return true;
+			} else if (FieldMap.TryGetValue (memberName, out var field)) {
+				member = field;
+				return true;
+			} else if (EventMap.TryGetValue (memberName, out var @event)) {
+				member = @event;
+				return true;
+			} else if (PropertyMap.TryGetValue (memberName, out var property)) {
+				member = property;
+				return true;
+			}
+			member = null;
+			return false;
```

Comment:
We can simplify the method a little and make it easier to expand if needed.
```suggestion
           foreach(var dict in new [] {MethodMap, FieldMap, EventMap, PropertyMap}) {
                      if (dict.TryGetValue (memberName, out member)) {
				                      return true;
				       }
           }
			member = null;
			return false;
```

Reply:
@stephen-hawley sorry for the indentation, github is not nice to write code.

Reply:
This can't be done - the dictionaries are different types so you can't make an array of them. And while each of them is type `IMemberDefinition`, I would prefer to keep them strongly typed. Also keep in mind that this API is likely to get called tens of millions of times and allocating an iterator and and array for each access is not ideal.